src/components/AnnotationsCarousel/AnnotationsCarousel.tsx

```diff
@@ -0,0 +1,85 @@
+import { Block, Elem } from '../../utils/bem';
+import { LsChevron } from '../../assets/icons';
+import { Button } from '../../common/Button/Button';
+import './AnnotationsCarousel.styl';
+import { AnnotationButton } from './AnnotationButton';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { observer } from 'mobx-react';
+import { clamp } from '../../utils/utilities';
+
+interface AnnotationsCarouselInterface {
+  store: any;
+  annotationStore: any; 
+  commentStore?: any;
+}
+
+export const AnnotationsCarousel = observer(({ store, annotationStore }: AnnotationsCarouselInterface) => {
+  const [entities, setEntities] = useState<any[]>([]);
+  const enableAnnotations = store.hasInterface('annotations:tabs');
+  const enablePredictions = store.hasInterface('predictions:tabs');
+  const enableCreateAnnotation = store.hasInterface('annotations:add-new');
+  const groundTruthEnabled = store.hasInterface('ground-truth');
+  const enableAnnotationDelete = store.hasInterface('annotations:delete');
+  const carouselRef = useRef<HTMLElement>();
+  const containerRef = useRef<HTMLElement>();
+  const [currentPosition, setCurrentPosition] = useState(0);
+  const [isLeftDisabled, setIsLeftDisabled] = useState(false);
+  const [isRightDisabled, setIsRightDisabled] = useState(false);
+
+  const updatePosition = useCallback((e: MouseEvent, goLeft = true) => {
+    if (containerRef.current && carouselRef.current) {
+      const step = containerRef.current.clientWidth;
+      const carouselWidth = carouselRef.current.clientWidth;
+      const newPos = clamp(goLeft ? currentPosition - step : currentPosition + step, 0, carouselWidth - step);
+
+      setCurrentPosition(newPos);
+    }
+  }, [containerRef, carouselRef, currentPosition]);
+  
+  useEffect(() => {
+    setIsLeftDisabled(currentPosition <= 0);
+    setIsRightDisabled(currentPosition >= ((carouselRef.current?.clientWidth ?? 0) - (containerRef.current?.clientWidth ?? 0)));
+  }, [containerRef.current, carouselRef.current, currentPosition, window.innerWidth, window.innerHeight]);
+
+  useEffect(() => {
+    const newEntities = [];
+
+    if (enablePredictions) newEntities.push(...annotationStore.predictions);
+  
+    if (enableAnnotations) newEntities.push(...annotationStore.annotations);
+    setEntities(newEntities);
```

Comment:
@Travis1282 I think here we need to ensure that the newEntities should be grouping the draft to always being first, otherwise there is a situation which I was able to replicate where the draft ends up elsewhere in the list, not at the beginning.

Reply:
@yyassi-heartex Just want to confirm this with you, but is it the intent that the draft does stick to the beginning, and all others are ordered as they exist in predictions/annotations?

Reply:
there is no real order for that - the code you're referring to was actually copy pasted from another spot where the list was being built in this way - so i dont think the order matters aslong as the icons are indicating things correctly